core/sql/sqlcomp/CmpSeabaseDDLrepos.cpp

```diff
@@ -358,6 +358,75 @@ short CmpSeabaseDDL::copyOldReposToNew(ExeCliInterface * cliInterface)
   return 0;
 }
 
+short CmpSeabaseDDL::migrateReposViews(ExeCliInterface * cliInterface,
+                                       NABoolean & someViewSaved /* out */)
+{
+  short retcode = 0;  // assume success
+  someViewSaved = FALSE;
+
+  // for each table that has been migrated, save and drop any views
+  // on the old table, and attempt to recreate them on the new
+
+  for (Int32 i = 0; i < sizeof(allReposUpgradeInfo)/sizeof(MDUpgradeInfo); i++)
+    {
+      const MDUpgradeInfo &rti = allReposUpgradeInfo[i];
+
+      if ((! rti.newName) || (! rti.oldName) || (NOT rti.upgradeNeeded))
+        continue;
+
+      Int64 tableUID = getObjectUID(cliInterface,
+        getSystemCatalog(), SEABASE_REPOS_SCHEMA, rti.oldName,
+        COM_BASE_TABLE_OBJECT_LIT, NULL, NULL, FALSE, FALSE /* ignore error */);
+
+      if (tableUID != -1)  // if we got it
+        {
+          NAList<NAString> viewNameList;
+          NAList<NAString> viewDefnList;
+ 
```

Comment:
If just the select text is saved, than any grants performed directly on the view would be lost.  For example, role admin_role is granted select on REPOS view one.  After upgrade, admin_role would no longer have the privilege and any user granted admin_role that runs a query referencing the view one will fail.  For a future upgrade, we may want handle privileges.